src/mudlet.cpp

```diff
@@ -691,16 +693,43 @@ mudlet::mudlet()
 //    });
 }
 
-QSettings* mudlet::getQSettings()
+static void migrateConfig(QSettings& settings) {
+    if (settings.contains("pos")) {
+        return;
+    }
+    // Old default configs, stored in NativeFormat
+    const QSettings settings_old1("mudlet", "Mudlet");
+    if (settings_old1.contains("pos")) {
+        for (auto& key : settings_old1.allKeys()) {
+            settings.setValue(key, settings_old1.value(key));
+        }
+        return;
+    }
+    const QSettings settings_old2("Mudlet", "Mudlet 1.0");
+    if (settings_old2.contains("pos")) {
+        for (auto& key : settings_old2.allKeys()) {
+            settings.setValue(key, settings_old2.value(key));
+        }
+        return;
+    }
+}
+
+void mudlet::setupConfig() {
+    if (QFile::exists(qsl("%1/portable.txt").arg(QCoreApplication::applicationDirPath()))) {
```

Comment:
I believe this should work fine on Windows. Not sure about macOS right now.

It doesn't work in Linux when running via an AppImage though (our official distribution method) since those get extracted to a temporary directory prior to running:

> access("/tmp/.mount_MudletSsu3Rv/portable.txt", F_OK) = -1 ENOENT (No such file or directory)

To solve this, have a look at the code [here](https://github.com/Mudlet/Mudlet/blob/development/src/updater.cpp#L291-L294) which figures out the location of the AppImage.